### PR TITLE
TDI-36651: adjust the return property for salesforce components

### PR DESCRIPTION
**Pull request by wwang-talend**
https://jira.talendforge.org/browse/TDI-36651

### CODE REVIEW

File: components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public SalesforceDefinition(String componentName) {
     @Override
     // Most of the components are on the input side, so put this here, the output definition will override this
     public Property[] getReturnProperties() {
-        return new Property[] { newInteger(RETURN_TOTAL_RECORD_COUNT) };
+        return new Property[] { newProperty(RETURN_ERROR_MESSAGE), newInteger(RETURN_TOTAL_RECORD_COUNT) };
```
Comment (gonchar-ivan) [inline components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java]: I am not sure we need to add ERROR_MESSAGE to components

Comment (wwang-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java]: The part also affect the ui part, so it's necessary now.

Comment (wwang-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java]: Please see the outline view in talend studio for salesforce components, it affect that.

Comment (wwang-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java]: In future, we can remove it and make it in framework part, but now just make it work.

Comment (gonchar-ivan) [inline components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java]: Wang Wei, thank you very much for explanation.
Do you know what class in Studio code is responsible for outline view?

Comment (wwang-talend) [inline components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java]: Please see this : org.talend.designer.core.generic.model.Component.createReturns()

Comment (sgandon) [inline components-salesforce/src/main/java/org/talend/components/salesforce/SalesforceDefinition.java]: I hate to see the same returns for every components, if all components need a ERROR_MESSAGE return then we can remove them all from components and change the studio to consider that all of the component have them.

**Pull request merged**